src/components/editMeasure/populationCriteria/groups/QDM/QDMMeasureGroups.tsx

```diff
@@ -449,6 +451,53 @@ const MeasureGroups = (props: MeasureGroupProps) => {
     }
   };
 
+  const validatePopulations = (
+    populations: Population[],
+    measureObservations: MeasureObservation[]
+  ): string => {
+    const returnTypesSet: Set<String> = new Set();
+    populations.forEach((population) => {
+      if (population.definition != undefined) {
+        const name = population.name;
+        const definition = population?.definition;
+        let newDefintion = "";
+        if (definition) {
+          let firstString = definition.substring(0, definition.indexOf(" "));
+          if (firstString) {
+            firstString = firstString.toLowerCase();
+          }
+          const secondString = definition.substring(
+            definition.indexOf(" ") + 1
+          );
+          if (secondString) {
+            secondString.replace(" ", "");
+          }
+          newDefintion = firstString + secondString;
+        }
+        const returnType = cqlDefinitionDataTypes[_.camelCase(newDefintion)];
+
+        if (returnType !== undefined) {
+          returnTypesSet.add(returnType);
+        }
+      }
+    });
+    if (measureObservations && measureObservations.length > 0) {
+      measureObservations.forEach((observation) => {
+        const returnType =
+          cqlDefinitionDataTypes[_.camelCase(observation.definition)];
+        if (returnType !== undefined) {
+          returnTypesSet.add(returnType);
+        }
+      });
+    }
```

Comment:
looks like we need to get return types multiple times for populations, observations, and maybe for stratifications in the future. we could write a general function to get the return types for selected defines?
e.g.
```
const getReturnTypes = (pops: Array<Population | MeasureObservation>) => {
    return pops.reduce((returnTypes, population) => {
      const definition = _.camelCase(_.trim(population?.definition));
      const returnType = cqlDefinitionDataTypes[definition];
      if (returnType) {
        returnTypes.push(returnType);
      }
      return returnTypes;
    }, []);
  };
```
which then can be called 

```
const populationReturnTypes = getReturnTypes(populations);
    const observationReturnTypes = getReturnTypes(measureObservations);
    const returnTypes = new Set([
      ...populationReturnTypes,
      ...observationReturnTypes,
    ]);
```

Reply:
@adongare This is great! A lot neater thanks so much Ashok! Just changed to use it.

Reply:
Sure, np